Domain/Services/Communication/XEntityResponse.cs

```diff
@@ -2,29 +2,23 @@
 
 namespace WebAPI.API.Domain.Services.Communication
 {
-    public class XEntityResponse : BaseResponse
+    public class XEntityResponse : BaseResponse<XEntity>
     {
-        public XEntity XEntity { get; private set; }
-
-        private XEntityResponse(bool success, string message, XEntity xEntity) : base(success, message)
-        {
-            XEntity = xEntity;
-        }
```

Comment:
because the base class now handles the object being passed out in the response, the derrived classes can be simpler by specifying the type T of the base class it inherits from.